### PR TITLE
Fix milk classification German

### PR DESCRIPTION
**Pull request by mahlzahn**
**Description:** Magermilch, entrahmte Milch = skimmed milk, fettarme Milch, teilentrahmte milch = semi-skimmed milk. See https://de.wikipedia.org/wiki/Milch#Fettgehalt

### CODE REVIEW

**Comment by mahlzahn** [comment]
Actually I wasn't done with it, yet. More changes follow soon here. (That's why PR is in draft state, thought it won't be published like this.)

File: taxonomies/ingredients.txt [hunk 1]
```diff
@@ -10418,6 +10602,7 @@ th:นมเปรี้ยว
 carbon_footprint_fr_foodges_ingredient:fr:Yaourt
 carbon_footprint_fr_foodges_value:fr:2.9
 vegetarian:en:yes
+vegan:en:no
```
Comment (mahlzahn) [inline taxonomies/ingredients.txt]:
```suggestion
```

**Comment by mahlzahn** [comment]
@aleene It's ready now. It includes quite some new terms (mainly for milk products) which help improving on issue #2231.

**Comment by aleene** [comment]
FYI: 
- When I add synonyms, I try check to put the name that seems to be most used on a product at the front.
- I update the wikidata entry with translations found on OFF, and add a link to the ingredient page on OFF.

File: taxonomies/ingredients.txt [hunk 2]
```diff
@@ -9954,7 +10106,7 @@ fr:lait de bufflonne pasteurisé
 
 <en:milk
 en:evaporated milk
-de:Kondensmilch
+de:evaporierte Milch, evaporierte Kondensmilch
```
Comment (aleene) [inline taxonomies/ingredients.txt]: Is `de:Kondensmilch` now deleted?

File: taxonomies/ingredients.txt [hunk 3]
```diff
@@ -10228,8 +10380,40 @@ fr:substances laitières modifiées
...
+es:Leche agria
+et:Hapupiim
+fr:lait fermenté, lait aigre, lait caillé
+is:Súrmjólk
+it:Latte acido
+ko:산유
+la:Oxygala
+lt:Rūgpienis
+mk:Кисело млеко
+nl:Dikmelk, gefermenteerde melk
```
Comment (aleene) [inline taxonomies/ingredients.txt]: nl:Dikmelk, Hangop, gefermenteerde melk

File: taxonomies/ingredients.txt (hunk 2 above)
Comment (mahlzahn) [reply taxonomies/ingredients.txt]: No, it was there twice, once as ```en:evaporated milk``` and once as ```en:condensed milk```.

File: taxonomies/ingredients.txt [hunk 4]
```diff
@@ -10228,8 +10380,40 @@ fr:substances laitières modifiées
...
+uk:Кисляк (молоко)
+zh_yue:酸奶
+zh:酸乳
+vegan:en:no
+vegetarian:en:yes
+wikipedia:en:https://en.wikipedia.org/wiki/Soured_milk
+wikidata:en:Q253333
 
 <en:milk
 fr:lait écrémé caillé
```
Comment (aleene) [inline taxonomies/ingredients.txt]: I am a bit lost on the fermented stuff. Wasn't there a section with de:Buttermilch. If so maybe it should be a child of "soured milk?

File: taxonomies/ingredients.txt [hunk 5]
```diff
@@ -9997,7 +10149,7 @@ az:Qatılaşdırılmış süd
 be:згушчанае малако
 ca:llet condensada
 cs:kondenzované mléko
-de:Kondensmilch, Dosenmilch, Büchsenmilch
+de:Kondensmilch, Dosenmilch, Büchsenmilch, eingedickte Milch
```
Comment (mahlzahn) [inline taxonomies/ingredients.txt]: Here it is still …

File: taxonomies/ingredients.txt (hunk 3 above)
Comment (mahlzahn) [reply taxonomies/ingredients.txt]: Will be in next commit.

File: taxonomies/ingredients.txt (hunk 5 above)
Comment (aleene) [reply taxonomies/ingredients.txt]: I remember now. There is a difference between evaporated and condensed milk (need to add some comments).

File: taxonomies/ingredients.txt (hunk 4 above)
Comment (mahlzahn) [reply taxonomies/ingredients.txt]: A nice overview over milk products in German can be found [here on wikipedia.org](https://de.wikipedia.org/wiki/Datei:Milch.svg).
```en:buttermilk``` is actually not just kind of soured milk. But we can add it as sub-category to the newly added category of ```en:fermented milk products```. There also other products could go, such as ```en:yogurt```, ```en:kefir```, ```en:soured milk```. @aleene Do you agree?

Comment (aleene) [reply taxonomies/ingredients.txt]: Seems logic. Go ahead.
We had a discussion on fermented/sour a while back: https://en.wiki.openfoodfacts.org/Creams_-_en. Name/definition also depends on country and local regulation. So be careful with combining "translations".

File: taxonomies/ingredients.txt (hunk 3 above)
Comment (mahlzahn) [reply taxonomies/ingredients.txt]: Done.

File: taxonomies/ingredients.txt (hunk 4 above)
Comment (mahlzahn) [reply taxonomies/ingredients.txt]: Done.

File: taxonomies/ingredients.txt [hunk 6]
```diff
@@ -8841,17 +8892,101 @@ wikipedia:en:https://en.wikipedia.org/wiki/Xylanase
...
 
-
-en:dairy, dairy product
-fr:produit laitier
-vegan:en:no
-
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 #
-#                                           B U T T E R     S E C T I O N
+#                               D A I R Y     P R O D U C T S     S E C T I O N
```
Comment (aleene) [inline taxonomies/ingredients.txt]: Note that these are not products, but ingredients. Products are found in the categories.

File: taxonomies/ingredients.txt [hunk 7]
```diff
@@ -8841,17 +8892,101 @@ wikipedia:en:https://en.wikipedia.org/wiki/Xylanase
...
+vegetarian:en:maybe
+
+<en:dairy
+en:powdered milk products, dried milk products, product on base on milk powder
+de:Trockenmilcherzeugnis
+
+
+################################ BUTTER related stuff ##########################
+
 # description:en:BUTTER is a dairy product with high butterfat content which is solid when chilled and at room temperature in some regions, and liquid when warmed. It is made by churning fresh or fermented cream or milk to separate the butterfat from the buttermilk. It is generally used as a spread on plain or toasted bread products and a condiment on cooked vegetables, as well as in cooking, such as baking, sauce making, and pan frying.
```
Comment (aleene) [inline taxonomies/ingredients.txt]: Ingredients and not products implies that all entries can be found on ingredients  lists. Is this the case for `en:fermented milk products`? If not, why introduce this ingredient? We do not want to redo the categories. The taxonomy tree should stay as flat as possible.

**Comment by aleene** [comment]
I have been gathering my thoughts on the ingredients taxonomy here, since I started with it. Feel free to edit and discuss. https://en.wiki.openfoodfacts.org/Project:Ingredients_ontology

File: taxonomies/ingredients.txt (hunk 6 above)
Comment (mahlzahn) [reply taxonomies/ingredients.txt]: Sure, but nevertheless, all milk products, categorised as ```en:dairy``` as main category.
Here I just made some improvements on the headings, which are just comments … Now it's as following:
```
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
#
#                               D A I R Y     P R O D U C T S     S E C T I O N
#
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
################################ BUTTER related stuff ##########################
################################## MILK related stuff ##########################
########################## LOW FAT and SKIMMED MILK ############################
################################# SEMI-SKIMMED MILK ############################
################################## FULL FAT MILKS ##############################
################################## RAW MILKS ###################################
################################## COW MILKS ###################################
################################# GOAT MILKS ###################################
################################# SHEEP MILKS ##################################
################################# BUFFALO MILKS ################################
################################# CONCENTRATED MILKS ###########################
################################ MILK POWDERS ##################################
######################### MILK and CREAM related stuff #########################
###################### LACTOSE and MINERALS related stuff ######################
################################## YOGURT related stuff ########################
################################## KEFIR related stuff #########################
################################## WHEY related stuff ##########################
################################## CREAM related stuff #########################
############################## SOURED MILK related stuff #######################
############################### BUTTERMILK related stuff #######################
```

File: taxonomies/ingredients.txt (hunk 7 above)
Comment (mahlzahn) [reply taxonomies/ingredients.txt]: At least in German it exists as ingredient (not very abundant), see https://world.openfoodfacts.org/ingredient/de:Sauermilcherzeugnis

File: taxonomies/ingredients.txt (hunk 6 above)
Comment (aleene) [reply taxonomies/ingredients.txt]: Was just to remind you. And indeed many ingredients can also be found as products in the categories taxonomy.

Comment (aleene) [reply taxonomies/ingredients.txt]: And en:dairy is actually found as ingredient, so we better support it well.

Comment (mahlzahn) [reply taxonomies/ingredients.txt]: Okay, thanks!

File: taxonomies/ingredients.txt (hunk 7 above)
Comment (mahlzahn) [reply taxonomies/ingredients.txt]: @aleene I'm fine also with reverting
```diff
-<en:fermented milk products
+<en:dairy
en:yogurt / en:kefir / en:soured milk / en:buttermilk
```
Do you prefer that?

**Comment by stephanegigandet** [comment]
Can we try to fix / merge this? Otherwise we'll probably get more conflicts and it will be harder.

File: taxonomies/ingredients.txt (hunk 7 above)
Comment (aleene) [reply taxonomies/ingredients.txt]: For the moment, yes

Comment (mahlzahn) [reply taxonomies/ingredients.txt]: Okay, done.

**Pull request merged**